pandas/core/indexes/datetimelike.py

```diff
@@ -69,7 +70,6 @@ class DatetimeIndexOpsMixin(ExtensionOpsMixin):
     _resolution = cache_readonly(DatetimeLikeArrayMixin._resolution.fget)
```

Comment:
So it looks like these `cache_readonly` calls are what's tripping up mypy. @jbrockmendel might be able to offer some insight here

Reply:
Is the problem with isort or mypy?

Reply:
If its mypy then would it be that it wants something like `# type: cache_readonly`?

Reply:
It's a mypy failure. Not actually with the cache_readonly as much as accessing the `.fget` attribute of the properties.

What's the intention of this code? Is passing fget from the property to cache_readonly faster than just accessing the property?

Reply:
> What's the intention of this code? Is passing fget from the property to cache_readonly faster than just accessing the property?

`cache_readonly behaves like `property`, expecting to get a unary method as its argument.  Does it even work if you don't make the `.fget` explicit?

Reply:
This failure is probably due to mypy. [https://github.com/python/mypy/issues/6185](url)

Deleting `.fget` run into other errors.